### PR TITLE
WX-1705 Support DRS Compact Identifiers

### PR DESCRIPTION
**Pull request by jgainerdewar**
This PR adds support for DRS compact identifiers, which are not valid IETL URIs. In order to allow the DRS URI transformation strategy to operate on them without requiring update of all transformation strategies, error handling has been added to `CombinedTransformationStrategy`.

This is in draft because I'll working through some errors caused by recent TES work.

### CODE REVIEW

File: src/TesApi.Tests/StartupTests.cs [hunk 1]
```diff
@@ -141,6 +141,34 @@ public void ConfigureServices_TerraOptionsAreNotConfigured_DefaultActionIdentity
             Assert.IsInstanceOfType(actionIdentityProvider, typeof(DefaultActionIdentityProvider));
         }
 
+        [TestMethod]
```
Comment (BMurri) [inline src/TesApi.Tests/StartupTests.cs:144]: I'm not sure if this was a bad merge or what, but it appears that the only changes in this file are these two methods are duplicated. Just revert this file, please.

**Comment by BMurri** [comment]
I'm wondering how this is expected to interact with #358

File: src/TesApi.Web/Management/Batch/MappingProfilePoolToWsmRequest.cs [hunk 2]
```diff
@@ -46,7 +48,12 @@ public MappingProfilePoolToWsmRequest()
             CreateMap<BatchVmContainerRegistry, ApiContainerRegistry>()
                 .ForMember(dst => dst.IdentityReference, opt => opt.MapFrom(src => src.IdentityResourceId));
             CreateMap<BatchNetworkConfiguration, ApiNetworkConfiguration>()
-                .ForPath(dst => dst.EndpointConfiguration.InboundNatPools, opt => opt.MapFrom(src => src.EndpointInboundNatPools));
+                .ForMember(dst => dst.EndpointConfiguration, opt => opt.MapFrom(src => src.EndpointInboundNatPools));
+            CreateMap<IList<BatchInboundNatPool>, ApiEndpointConfiguration>()
+                .ForMember(dst => dst.InboundNatPools, opt => opt.MapFrom(src => src))
+                .ReverseMap();
+            CreateMap<ApiInboundNatPool[], IList<BatchInboundNatPool>>()
+                .ForMember("Item", opt => opt.Ignore());
```
Comment (jgainerdewar) [inline src/TesApi.Web/Management/Batch/MappingProfilePoolToWsmRequest.cs:56]: This is a fix for a problem with the shape of the WSM batch pool creation request.

Comment (BMurri) [reply src/TesApi.Web/Management/Batch/MappingProfilePoolToWsmRequest.cs:56]: That same fix is in #764

**Comment by BMurri** [comment]
Replacing this PR with #766

**Pull request closed without merging**